packages/twenty-server/src/engine/metadata-modules/remote-server/remote-table/remote-table.service.ts

```diff
@@ -445,33 +393,37 @@ export class RemoteTableService {
       workspaceId: workspaceId,
       icon: 'IconPlug',
       isRemote: true,
-      primaryKeyColumnType: remoteTableIdColumn.udtName,
-      // TODO: function should work for other types than Postgres
-      primaryKeyFieldMetadataSettings: mapUdtNameToSettings(
-        remoteTableIdColumn.udtName,
+      primaryKeyColumnType: distantTableIdColumn.udtName,
+      primaryKeyFieldMetadataSettings: mapUdtNameToFieldSettings(
+        distantTableIdColumn.udtName,
       ),
     } satisfies CreateObjectInput);
 
-    for (const column of remoteTableColumns) {
-      const field = await this.fieldMetadataService.createOne({
-        name: column.columnName,
-        label: camelToTitleCase(camelCase(column.columnName)),
-        description: 'Field of remote',
-        // TODO: function should work for other types than Postgres
-        type: mapUdtNameToFieldType(column.udtName),
-        workspaceId: workspaceId,
-        objectMetadataId: objectMetadata.id,
-        isRemoteCreation: true,
-        isNullable: true,
-        icon: 'IconPlug',
-        // TODO: function should work for other types than Postgres
-        settings: mapUdtNameToSettings(column.udtName),
-      } satisfies CreateFieldInput);
-
-      if (column.columnName === 'id') {
-        await this.objectMetadataService.updateOne(objectMetadata.id, {
-          labelIdentifierFieldMetadataId: field.id,
-        });
+    for (const column of distantTableColumns) {
+      // TODO: return error to the user when a column cannot be managed
+      try {
+        const field = await this.fieldMetadataService.createOne({
+          name: column.columnName,
```

Comment:
To check if this works with column names with snake_case, I believe this is rejected as of today

Reply:
I updated the logic to use camelCase !